rpc/transaction.go

```diff
@@ -21,17 +21,20 @@ func adaptTransaction(t TXN) (Transaction, error) {
 		json.Unmarshal(txMarshalled, &tx)
 		return tx, nil
 	case TransactionType_Declare:
+
 		switch t.Version {
-		case new(felt.Felt).SetUint64(1):
+		case &felt.Zero:
+			var tx DeclareTxnV0
+			json.Unmarshal(txMarshalled, &tx)
+			return tx, nil
+		case felt.Zero.SetUint64(1):
 			var tx DeclareTxnV1
 			json.Unmarshal(txMarshalled, &tx)
 			return tx, nil
-		case new(felt.Felt).SetUint64(2):
+		case felt.Zero.SetUint64(2):
```

Comment:
`Set*` methods in felt first change value inside pointer, then return it's pointer.
After calling `felt.Zero.SetUint64(...)` here and above `felt.Zero` contains either 1 or 2.

My suggestion is to setup known versions before hand:
```
var (
    Version0 = felt.Zero
    Version1 = new(felt).SetUint(1)
    Version2 = ...
)
```

Also please note that right now is pointer comparison, not value comparison IMO it's safer to check like through `felt.Equal()`:
switch {
  case t.Version.Equal(Version1):
      ...
  case t.Version.Equal(Version2):
    ....
}
```